umap/static/umap/js/umap.controls.js

```diff
@@ -1252,16 +1252,37 @@ L.U.Editable = L.Editable.extend({
         var content;
         if (e.layer instanceof L.Marker) content = L._('Click to add a marker');
         else if (e.layer instanceof L.Polyline) {
-            if (!e.layer.editor._drawnLatLngs.length) {
-                if (e.layer instanceof L.Polygon) content = L._('Click to start drawing a polygon');
-                else if (e.layer instanceof L.Polyline) content = L._('Click to start drawing a line');
-            } else if (e.layer.editor._drawnLatLngs.length < e.layer.editor.MIN_VERTEX) {
-                content = L._('Click to continue drawing');
+            // when drawing a Polyline or Polygon
+            if (e.layer.editor._drawnLatLngs) {
+                // when drawing first point
+                if (!e.layer.editor._drawnLatLngs.length) {
+                    if (e.layer instanceof L.Polygon){
+                        content = L._('Click to start drawing a polygon');
+                    } else if (e.layer instanceof L.Polyline) {
+                        content = L._('Click to start drawing a line');
+                    }
+                } else {
+                    var readableDistance = e.layer.getMeasure(e.latlng);
+                    if (e.layer.editor._drawnLatLngs.length < e.layer.editor.MIN_VERTEX) {
+                        // when drawing second point
+                        content = L._('Click to continue drawing ({distance})', { distance: readableDistance });
+                    }  else {
+                        // when drawing third point (or more)
+                        content = L._('Click last point to finish shape ({distance})', { distance: readableDistance });
+                    }
+                }
             } else {
-                content = L._('Click last point to finish shape');
+                // when moving an existing point
+                if (e.layer instanceof L.Polygon){
+                    content = L._('Polygon area: {area}', { area: e.layer.getMeasure() });
```

Comment:
Maybe we could always keep this phrase, and when drawing append it after the "Click blah blah", what do you think ?

Reply:
(And in this case you don't have to deal with translation)

Reply:
> Ah, it seems that now the measure is not update _while_ drawing the last point (i.e. it does not includes the currently moving point), is that on purpose ?

It does _not_? When i create a new polyline, on my end it looks like this (at least in Firefox and Safari):
<img width="800" alt="Bildschirmfoto 2023-04-27 um 11 01 48" src="https://user-images.githubusercontent.com/1724290/234813890-1256475e-eb3c-46b4-9643-f2da72f5f023.png">